commit 02cc60f6935a5005aa461da183c6c4332503be83
Author: Ilsoo Byun <ilsoobyun@linecorp.com>
Date:   Tue Sep 10 18:13:02 2019 +0900

    mgr: do not reset reported if a new metric is not collected
    
    Fixes: https://tracker.ceph.com/issues/41741
    Signed-off-by: Ilsoo Byun <ilsoobyun@linecorp.com>

diff --git a/src/mgr/DaemonHealthMetricCollector.h b/src/mgr/DaemonHealthMetricCollector.h
index 731b4c43b5c..42bf905fa9b 100644
--- a/src/mgr/DaemonHealthMetricCollector.h
+++ b/src/mgr/DaemonHealthMetricCollector.h
@@ -12,7 +12,7 @@ public:
   static std::unique_ptr<DaemonHealthMetricCollector> create(daemon_metric m);
   void update(const DaemonKey& daemon, const DaemonHealthMetric& metric) {
     if (_is_relevant(metric.get_type())) {
-      reported = _update(daemon, metric);
+      reported |= _update(daemon, metric);
     }
   }
   void summarize(health_check_map_t& cm) {
